### PR TITLE
fix(integration): create a new test dir per language

### LINKED ISSUE #2412: Multi language tests should use a new directory every time
Currently state from the previous language can interfere with the following tests.

### PR DESCRIPTION
**Pull request by alecthomas**
Fixes #2412

### CODE REVIEW

File: jvm-runtime/java/scaffolding/go.mod [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module ignore
+
+go 1.22.6
```
Comment (stuartwdouglas) [inline jvm-runtime/java/scaffolding/go.mod:3]: Just out of curiosity why is this necessary? I understand why you need it for the go scaffolding but I assumed this would not be needed for Java as there are no go files in the directory?

**Review by stuartwdouglas — APPROVED** [review]

File: jvm-runtime/java/scaffolding/go.mod (hunk 1 above)
Comment (alecthomas) [reply jvm-runtime/java/scaffolding/go.mod:3]: I ended up fixing this separately #2414, I explained it in a bit more detail there.
